src/hotspot/share/prims/jvmtiExport.cpp

```diff
@@ -198,12 +198,8 @@ class JvmtiVirtualThreadEventMark : public JvmtiEventMark {
 public:
   JvmtiVirtualThreadEventMark(JavaThread *thread) :
     JvmtiEventMark(thread) {
-    JvmtiThreadState* state = thread->jvmti_thread_state();
-    if (state != NULL && state->is_virtual()) {
-      _jthread = to_jobject(thread->vthread());
-    } else {
-      _jthread = to_jobject(thread->threadObj());
-    }
+    _jthread = to_jobject(thread->vthread());
+    assert(thread->vthread() != NULL || thread->threadObj() == NULL, "sanity check");
```

Comment:
Seems a little strange to me that L202 is after L201. You're asserting
that `thread->vthread() != NULL` after passing it to a `to_jobject()` call.

Reply:
I do not think this matters.
But I will revert the order of these line to make you happy. :)
Thank you for looking at the fix!